src/measure_theory/integral/bochner.lean

```diff
@@ -293,77 +331,42 @@ end
 
 lemma integral_add {f g : α →ₛ E} (hf : integrable f μ) (hg : integrable g μ) :
   integral μ (f + g) = integral μ f + integral μ g :=
-calc integral μ (f + g) = ∑ x in (pair f g).range,
-       ennreal.to_real (μ ((pair f g) ⁻¹' {x})) • (x.fst + x.snd) :
-begin
-  rw [add_eq_map₂, map_integral (pair f g)],
-  { exact integrable_pair hf hg },
-  { simp only [add_zero, prod.fst_zero, prod.snd_zero] }
-end
-... = ∑ x in (pair f g).range,
-        (ennreal.to_real (μ ((pair f g) ⁻¹' {x})) • x.fst +
-         ennreal.to_real (μ ((pair f g) ⁻¹' {x})) • x.snd) :
-  finset.sum_congr rfl $ assume a ha, smul_add _ _ _
-... = ∑ x in (pair f g).range,
-        ennreal.to_real (μ ((pair f g) ⁻¹' {x})) • x.fst +
-      ∑ x in (pair f g).range,
-        ennreal.to_real (μ ((pair f g) ⁻¹' {x})) • x.snd :
-  by rw finset.sum_add_distrib
-... = ((pair f g).map prod.fst).integral μ + ((pair f g).map prod.snd).integral μ :
-begin
-  rw [map_integral (pair f g), map_integral (pair f g)],
-  { exact integrable_pair hf hg }, { refl },
-  { exact integrable_pair hf hg }, { refl }
-end
-... = integral μ f + integral μ g : rfl
+set_to_simple_func_add _ weighted_smul_union hf hg
 
 lemma integral_neg {f : α →ₛ E} (hf : integrable f μ) : integral μ (-f) = - integral μ f :=
-calc integral μ (-f) = integral μ (f.map (has_neg.neg)) : rfl
-  ... = - integral μ f :
-  begin
-    rw [map_integral f _ hf neg_zero, integral, ← sum_neg_distrib],
-    refine finset.sum_congr rfl (λx h, smul_neg _ _),
-  end
+set_to_simple_func_neg _ weighted_smul_union hf
 
-lemma integral_sub [borel_space E] {f g : α →ₛ E} (hf : integrable f μ) (hg : integrable g μ) :
+lemma integral_sub {f g : α →ₛ E} (hf : integrable f μ) (hg : integrable g μ) :
   integral μ (f - g) = integral μ f - integral μ g :=
-begin
-  rw [sub_eq_add_neg, integral_add hf, integral_neg hg, sub_eq_add_neg],
-  exact hg.neg
-end
+set_to_simple_func_sub _ weighted_smul_union hf hg
 
 lemma integral_smul (c : 𝕜) {f : α →ₛ E} (hf : integrable f μ) :
   integral μ (c • f) = c • integral μ f :=
-calc integral μ (c • f) = ∑ x in f.range, ennreal.to_real (μ (f ⁻¹' {x})) • c • x :
-  by rw [smul_eq_map c f, map_integral f _ hf (smul_zero _)]
-... = ∑ x in f.range, c • (ennreal.to_real (μ (f ⁻¹' {x}))) • x :
-  finset.sum_congr rfl $ λ b hb, by { exact smul_comm _ _ _}
-... = c • integral μ f :
-by simp only [integral, smul_sum, smul_smul, mul_comm]
+set_to_simple_func_smul _ weighted_smul_union weighted_smul_smul c hf
+
+lemma norm_set_to_simple_func_le_integral_norm (T : set α → E →L[ℝ] F) {C : ℝ}
+  (hT_norm : ∀ s, ∥T s∥ ≤ C * (μ s).to_real) {f : α →ₛ E} (hf : integrable f μ) :
+  ∥f.set_to_simple_func T∥ ≤ C * (f.map norm).integral μ :=
+calc ∥f.set_to_simple_func T∥
+    ≤ C * ∑ x in f.range, ennreal.to_real (μ (f ⁻¹' {x})) * ∥x∥ :
+  norm_set_to_simple_func_le_sum_mul_norm T hT_norm f
+... = C * (f.map norm).integral μ : by { rw map_integral f norm hf norm_zero, simp_rw smul_eq_mul, }
```

Comment:
can you move this?

Reply:
It requires the integral to be defined. I cannot put it in the other file.

Reply:
Ha, yes, sorry!